extensions/binding/org.eclipse.smarthome.binding.sonos/src/main/java/org/eclipse/smarthome/binding/sonos/internal/handler/ZonePlayerHandler.java

```diff
@@ -683,10 +683,18 @@ protected void updateChannel(String channeldD) {
             case CURRENTALBUMART:
                 url = getAlbumArtUrl();
                 if (url != null) {
-                    RawType image = HttpUtil.downloadImage(url, true, 500000);
-                    if (image != null) {
-                        newState = image;
-                    }
+                    // We run the update of the covert art channel in a different thread
+                    // because it can take time
+                    newState = null;
+                    scheduler.schedule(new Runnable() {
```

Comment:
1. please use "submit" instead of schedule
2. please use Java8 syntax and not anonymous inner classes